AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin+GraphQLModelBehavior.swift

```diff
@@ -12,7 +12,7 @@ public extension AWSAPIPlugin {
     func query<M: Model>(from modelType: M.Type,
                          byId id: String,
                          listener: GraphQLOperation<M?>.EventListener?) -> GraphQLOperation<M?> {
-        let request = GraphQLRequest<M>.query(from: modelType, byId: id)
+        let request = GraphQLRequest<M?>.query(from: modelType, byId: id)
```

Comment:
Sorry, need to ask.. what does adding a `?` to the type here do, and why are we adding it?

Reply:
> Sorry, need to ask.. what does adding a `?` to the type here do, and why are we adding it?

1. "What": GraphQLRequest is a generic type whose type arg specifies the decoded response type. For model-based GraphQL APIs, we can do things like `GraphQLRequest<Post>` to specify that the request's response should be decoded into a single `Post` object, or `GraphQLRequest<[Post]>` to specify an array of `Post` objects, etc. In this case, `GraphQLRequest<M?>` says the response type is an Optional `M`, meaning the query result could be either an `M` or nil. 
1. "Why": A query isn't guaranteed to have a result. For "byId" queries, we know that a valid result will only ever have zero or one object, so we can specify the response decodes to an `Optional<M>` to accurately reflect that expectation. (In contrast, we can't predict how many results a predicate query might have, so we specify that result as an array. We *could* return `nil` for a no-result case, but instead we specify a return type of `[M]` and return an empty array, making it convenient to do things like `for foo in result { }` without having to guard for nil results.

Reply:
+100, thanks tim.